erpnext/projects/doctype/project/project.py

```diff
@@ -386,6 +387,27 @@ def set_gross_margin(self, update=False, update_modified=False):
 				'per_gross_margin': self.per_gross_margin,
 			}, None, update_modified=update_modified)
 
+	def set_tasks_status(self, update=False, update_modified=False):
+		tasks_data = frappe.get_all("Task", fields=["name", "status"], filters={
+			"project": self.name,
+			"status": ["!=", "Cancelled"],
+		})
+
+		if tasks_data:
+			if all(d.status == "Completed" for d in tasks_data):
+				self.tasks_status = "Completed"
+			elif any(d.status in ["Working", "Pending Review"] for d in tasks_data):
+				self.tasks_status = "In Progress"
+			elif any(d.status == "On Hold" for d in tasks_data) and all(d.status != "Working" for d in tasks_data):
+				self.tasks_status = "On Hold"
+			elif all(d.status == "Open" for d in tasks_data):
+				self.tasks_status = "Not Started"
```

Comment:
These conditions should have an else block to handle any future cases. The "In Progress" part should be in the else block and everything else should have an explicit condition

Reply:
CHeck for Working and Pending review both in On Hold condition